src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1495,6 +1495,45 @@
       (sql/do-commands
        "ALTER TABLE certnames ADD CONSTRAINT certnames_reports_id_fkey FOREIGN KEY (latest_report_id) REFERENCES reports(id) ON DELETE SET NULL")))
 
+
+(defn change-value-integer-to-numeric
+  []
+  (let [hash-type (if (sutils/postgres?) "bytea" "varchar(40)")
+        munge-hash (if (sutils/postgres?)
+                     (fn [column] (format "('\\x' || %s)::bytea" column)) identity)]
+    (sql/create-table :fact_values_transform
+                      [ "id" "bigint NOT NULL DEFAULT nextval('fact_values_id_seq')"]
+                      ["value_hash" hash-type "NOT NULL"]
+                      ["value_type_id" "bigint NOT NULL"]
+                      ["value_integer numeric(1000,0)"]
+                      ["value_float" "double precision"]
```

Comment:
It might sound looney, but this one is possibly going to be a problem too, and it will fall into the same bucket. Thinking ahead, is this theoretical outcome just going to force us into 1 column? I'm just conjecturing, I'm not sure we should change it just yet. One interesting side-effect will be a less complicated cast methodology.